internal/pkg/cache/cache.go

```diff
@@ -0,0 +1,69 @@
+package cache
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+var (
+	cacheFolderPath string
+
+	identifierRegex             = regexp.MustCompile("^[a-zA-Z0-9-]+$")
+	ErrorInvalidCacheIdentifier = fmt.Errorf("invalid cache identifier")
+)
+
+func init() {
```

Comment:
Can you move this to a `Init` function and call it on the `login` command? Then you can pass the `cacheFolderPath` as argument to the other functions